components/rollup_node/block_gen.md

```diff
@@ -0,0 +1,147 @@
+# Block Generation
+
+The logic which is used to generate the rollup chain from L1.
+
+## Glossary
+
+- **Block Inputs**: All information required to generate a full Ethereum block, given the pre-state and previous block. For example, does NOT include a state root for that block's input.
+- **Rollup Blockchain**: The rollup blocks generated by performing the rollup's state transition function over all block inputs. For example, DOES include a state root for that block's input.
+
+## Summary
+
+The rollup chain can be deterministically generated given an L1 Ethereum chain. The fact that the entire rollup chain can be generated based on L1 blocks is _what makes OE a rollup_. This process can be represented as:
+
+```python
+f(l1_blockchain) -> rollup_blockchain
+```
+
+In this document we define a block generation function which is designed to be:
+
+1. Context independent given a constant sized subset of the L1 chain.
+2. Support sequencers and sequencer consensus.
+3. Resilient sequencer censorship.
+
+## Rollup Epochs
+
+The rollup chain is subdivided into epochs. There is a 1:1 correspondence between L1 block numbers and epoch numbers. For L1 block number `n`, there is a corresponding rollup epoch `n` which can only be generated after L1 block number `n` is added to the L1 chain. Epoch contain one to many rollup blocks.
+
+### Types of Blocks within Epochs
+
+Within these epochs, there are two block types:
+
+1. Deposit block
+2. Sequencer block
+
+Deposit blocks contain contextual information about L1 such as the block hash. They also contain transactions initiated on L1 by any user or contract.
```

Comment:
`... on L1 by any user or contract that execute on L2` or something along those lines

Reply:
Right now ETH deposits are a special kind of "enqueued" transaction. We may consider in the future making ETH deposits special (not require paying L2 gas) and then have another type of more generic deposit which actually has to pay gas on L2. But right now everything goes through the "enqueue" pathway.

Also, in this spec we're trying to move away from the "enqueue" term and towards "deposit" but you'll still notice enqueue terminology is used often in our current docs & convos.